hail/python/hailtop/gear/deploy_config.py

```diff
@@ -0,0 +1,57 @@
+import os
+
+class DeployConfig:
+    def __init__(self, *, location='external', config=None, config_file=None):
+        self.location = location
+        if not config:
+            if not config_file:
+                for f in [os.path.expanduser('~/.hail/deploy_config.json'),
+                          '/deploy-config/deploy-config.json']:
+                    if os.path.exists(f):
+                        config_file = f
+                        break
+
+            if config_file:
+                with open(config_file, 'r') as f:
+                    config = json.loads(f.read())
+            else:
+                config = {
+                    'default_namespace': 'default',
+                    'service_namespace': {}
+                }
+
+        self._default_ns = config['default_namespace']
+        self._service_namespace = config['service_namespace']
+
+    def _service_ns(self, service):
+        return self._service_namespace.get(service, self._default_ns)
+
+    def scheme(self, location=None):
+        if not location:
+            location = self.location
+        return 'https' if location == 'external' else 'http'
+
+    def domain(self, service, location=None):
+        if not location:
+            location = self.location
+        ns = self._service_ns(service)
```

Comment:
should this be _service_namespace?

Reply:
Sorry, my names aren't great.  `default_namespace` is the namespace to expect a given service in by default.  `service_namespace` is a set of overrides for specific services.  I want to support that so I can e.g. deploy batch in my cseed, but use the auth service in default.  Make sense?  So everyone should go through _service_ns.